ts/spi/provider_frame.ts

```diff
@@ -460,7 +460,7 @@ export class ProviderFrame {
       Promise<Credential[]> {
     // get all credentials across all domains; from this, we can filter down
     // to the set of credentials
-    let allCredentials = await this.credentialDataProvider.getAllCredentials();
+    let allCredentials = await this.credentialDataProvider.getAllHints(options);
```

Comment:
Yes, I agree. By removing work from providers we also made many decisions for them. Not to do in this PR, but add a TODO to extract the de-duplication code?